### DISCUSSION TITLE
How do modify dict elmement？

### DISCUSSION THREAD

**Discussion by acekingke** [hylang/hy#2435] [Q&A] [answered]
Given dict in python` obj = {"a": {"c":1}, "b":2} ` , I want  `obj["a"]["c"]=2` , but how to do it in hylang?

**Comment by Kodiologist** [comment]
I bet you can figure this one out on your own pretty easily. Consider: how do you write `c = 2` in Hy? How do you write `obj["a"]` in Hy? Try building off from there. Let me know if you get stuck.

**Reply by acekingke** [reply]
yes, 
```hy
(setv obj {"a" {"c" 1} "b" 2})
```
how to change obj key c 's value in hy.

```

**Reply by Kodiologist** [reply, marked as answer]
Try `(setv (get obj "a" "c") 2)`.

**Reply by acekingke** [reply]
Thank you very much

**Comment by goyalyashpal** [comment]
I was also trying to use dictionary, but couldn't figure out the equivalent of python dict's `a['3']` in hy
i was referring this hy tutorial: https://docs.hylang.org/en/stable/tutorial.html
& the `get` operation referred to there: https://docs.hylang.org/en/stable/api.html#hy.pyops.get


```hy
(setv a #{1 2  3 4})

a
  ; #{1 2 3 4}

(print a)
  ; {1, 2, 3, 4}
```

```hy
=> (get a 3)
Traceback (most recent call last):
  File "stdin-8d3522acff2f2f4b569e9551e76ef8e5bbb56269", line 1, in <module>
TypeError: 'set' object is not subscriptable
=> (get a "3")
Traceback (most recent call last):
  File "stdin-ee295049c0d832ca7a5417ea612647e2a70c6382", line 1, in <module>
TypeError: 'set' object is not subscriptable
```

----
Also note the weird order here: `"1" 2 "3" 4` -> `"3" 2 "1" 4`

```hy
(setv a #{"1" 2  "3" 4})

a
  ; #{"3" 2 "1" 4}

(print a)
  ; {'3', 2, '1', 4}
```

```hy
=> (print (get a "3"))
Traceback (most recent call last):
  File "stdin-8b322e50ba4b8c8af946bb5553aa07dacbdd0e56", line 1, in <module>
TypeError: 'set' object is not subscriptable
```

**Reply by Kodiologist** [reply]
`set` and `dict` are different types, with different literal syntax.

**Reply by goyalyashpal** [reply]
> `set` and `dict` are different types, with different literal syntax.

yes; did i accidenlty mix those? per the [tutorial], i don't think so. but obviously i am missing smth

[tutorial]: https://docs.hylang.org/en/stable/tutorial.html#literals


<img src="https://github.com/hylang/hy/assets/19423063/21fd1b52-4c79-41bd-abca-9c6ffd222f8c" width=400px/>

**Reply by Kodiologist** [reply]
Yep. Look how you set `a` above.

**Reply by goyalyashpal** [reply]
oh, sorry 😆 - i mingled it while reading.

```hy
(setv a {"1" 2 "3" 4})
(get a "3")
  ; 4
```

worked 👍
